pkg/clusteragent/clusterchecks/dispatcher_configs.go

```diff
@@ -91,3 +91,37 @@ func (d *dispatcher) retrieveAndClearDangling() []integration.Config {
 	danglingConfigs.Set(0)
 	return configs
 }
+
+// patchConfiguration transforms the configuration from AD into a config
+// ready to use by node agents. It does the following changes:
+//   - empty the ADIdentifiers array, to avoid node-agents detecting them as templates
+//   - clear the ClusterCheck boolean
+//   - add the empty_default_hostname option to all instances
+//   - inject the extra tags (including `cluster_name` if set) in all instances
+func (d *dispatcher) patchConfiguration(in integration.Config) (integration.Config, error) {
+	out := in
+	out.ADIdentifiers = nil
+	out.ClusterCheck = false
```

Comment:
Why do we do this? Don't we want to be able to identify these? 🤔

Reply:
The check collector [ignores configurations with that flag](https://github.com/DataDog/datadog-agent/blob/fb6f5e51e2450d577b3cb8414c6ccacab0887f75/pkg/collector/scheduler.go#L199-L202). This is needed to avoid having them run in the DCA's collector.

The patching logic is taking a cluster-check configuration and outputting a "classic" check configuration ready to run on a node.

We are able to identify them on the node-agent's `configcheck` via their source (the `clusterchecks` config provider), see an example output in https://docs-staging.datadoghq.com/xvello/cchecks/agent/autodiscovery/clusterchecks/#autodiscovery-in-the-node-based-agent